File: Release/Silverlight5/SampleSource/Controls/Controls.Samples/Charting/Data/Pair.cs
// (c) Copyright Microsoft Corporation.
// This source is subject to the Microsoft Public License (Ms-PL).
// Please see http://go.microsoft.com/fwlink/?LinkID=131993 for details.
// All other rights reserved.

using System.ComponentModel;

namespace System.Windows.Controls.Samples
{
    /// <summary>
    /// Class representing an untyped pair of values.
    /// </summary>
    public class Pair : INotifyPropertyChanged
    {
        /// <summary>
        /// Gets or sets the first value.
        /// </summary>
        public object First
        {
            get
            {
                return _first;
            }
            set
            {
                _first = value;
                OnPropertyChanged("First");
            }
        }

        /// <summary>
        /// Stores the value of the First property.
        /// </summary>
        private object _first;

        /// <summary>
        /// Gets or sets the second value.
        /// </summary>
        public object Second
        {
            get
            {
                return _second;
            }
            set
            {
                _second = value;
                OnPropertyChanged("Second");
            }
        }

        /// <summary>
        /// Stores the value of the Second property.
        /// </summary>
        private object _second;

        /// <summary>
        /// Implements the INotifyPropertyChanged interface.
        /// </summary>
        public event PropertyChangedEventHandler PropertyChanged;

        /// <summary>
        /// Fires the PropertyChanged event.
        /// </summary>
        /// <param name="propertyName">Name of the property that changed.</param>
        private void OnPropertyChanged(string propertyName)
        {
            PropertyChangedEventHandler handler = PropertyChanged;
            if (null != handler)
            {
                handler.Invoke(this, new PropertyChangedEventArgs(propertyName));
            }
        }
    }
}

File: Release/Silverlight5/SampleSource/Controls/Controls.Samples/Charting/Data/Pair.vb
' (c) Copyright Microsoft Corporation.
' This source is subject to the Microsoft Public License (Ms-PL).
' Please see http://go.microsoft.com/fwlink/?LinkID=131993 for details.
' All other rights reserved.

Imports Microsoft.VisualBasic
Imports System.ComponentModel

''' <summary>
''' Class representing an untyped pair of values.
''' </summary>
Public Class Pair
    Implements INotifyPropertyChanged
    ''' <summary>
    ''' Gets or sets the first value.
    ''' </summary>
    Public Property First() As Object
        Get
            Return _first
        End Get
        Set(ByVal value As Object)
            _first = value
            OnPropertyChanged("First")
        End Set
    End Property

    ''' <summary>
    ''' Stores the value of the First property.
    ''' </summary>
    Private _first As Object

    ''' <summary>
    ''' Gets or sets the second value.
    ''' </summary>
    Public Property Second() As Object
        Get
            Return _second
        End Get
        Set(ByVal value As Object)
            _second = value
            OnPropertyChanged("Second")
        End Set
    End Property

    ''' <summary>
    ''' Stores the value of the Second property.
    ''' </summary>
    Private _second As Object

    ''' <summary>
    ''' Implements the INotifyPropertyChanged interface.
    ''' </summary>
    Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

    ''' <summary>
    ''' Fires the PropertyChanged event.
    ''' </summary>
    ''' <param name="propertyName">Name of the property that changed.</param>
    Private Sub OnPropertyChanged(ByVal propertyName As String)
        Dim handler As PropertyChangedEventHandler = PropertyChangedEvent
        If Nothing IsNot handler Then
            handler.Invoke(Me, New PropertyChangedEventArgs(propertyName))
        End If
    End Sub
End Class
